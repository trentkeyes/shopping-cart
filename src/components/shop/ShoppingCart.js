import React from 'react';
import '../../styles/shoppingCart.css';

export const ShoppingCart = ({ cart }) => {
  const cartArr = Object.values(cart);

  const cartElements = cartArr.map((item, index) => {
    const album = item.info;
    return (
      <div key={index}>
        <div className="shopping-cart__flex">
          <p>
            {album.album} <span>({album.artist})</span>
          </p>
          <p className="shopping-cart__push-right">
            ({item.quantity})
            <span className="underline">${item.quantity * album.price}</span>
          </p>
        </div>
      </div>
    );
  });

  const totals = cartArr.reduce(
    (acc, current) => {
      const item = current.info;
      const quant = current.quantity;
      console.log(item);
      return {
        quantity: acc.quantity + quant,
        price: Math.round(acc.price + quant * item.price),
      };
    },
    { quantity: 0, price: 0 }
  );
  return (
    <div>
      {cartArr.length > 0 ? (
        <div className="shopping-cart">
          <div>{cartElements}</div>
          <div className="shopping-cart__flex shopping-cart__total">
            <p className="">Total</p>
            <p className="shopping-cart__push-right">
              <span>({totals.quantity})</span>
              <span className="underline"> ${totals.price}</span>
            </p>
          </div>
        </div>
      ) : (
        <p>Add something to your cart</p>
      )}
    </div>
  );
};
