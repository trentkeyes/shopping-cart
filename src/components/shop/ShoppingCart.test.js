import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { ShoppingCart } from './ShoppingCart';

const cart = {
  0: {
    quantity: 1,
    info: {
      album: 'Album 1',
      artist: 'Artist 1',
      price: 10,
    },
  },
  1: {
    quantity: 2,
    info: {
      album: 'Album 2',
      artist: 'Artist 2',
      price: 20,
    },
  },
};

test('ShoppingCart component renders correctly with items in the cart', () => {
  const { container } = render(<ShoppingCart cart={cart} />);
  expect(container).toMatchSnapshot();
});

test('ShoppingCart component shows the correct total quantity and price', () => {
  const { getByText } = render(<ShoppingCart cart={cart} />);
  expect(getByText('(3)')).toBeInTheDocument();
  expect(getByText('$50')).toBeInTheDocument();
});

test('ShoppingCart component shows the correct message when the cart is empty', () => {
  const { getByText } = render(<ShoppingCart cart={{}} />);
  expect(getByText('Add something to your cart')).toBeInTheDocument();
});
