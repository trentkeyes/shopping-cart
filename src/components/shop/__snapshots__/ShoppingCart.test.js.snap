// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingCart component renders correctly with items in the cart 1`] = `
<div>
  <div>
    <div
      class="shopping-cart"
    >
      <div>
        <div>
          <div
            class="shopping-cart__flex"
          >
            <p>
              Album 1
               
              <span>
                (
                Artist 1
                )
              </span>
            </p>
            <p
              class="shopping-cart__push-right"
            >
              (
              1
              )
              <span
                class="underline"
              >
                $
                10
              </span>
            </p>
          </div>
        </div>
        <div>
          <div
            class="shopping-cart__flex"
          >
            <p>
              Album 2
               
              <span>
                (
                Artist 2
                )
              </span>
            </p>
            <p
              class="shopping-cart__push-right"
            >
              (
              2
              )
              <span
                class="underline"
              >
                $
                40
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="shopping-cart__flex shopping-cart__total"
      >
        <p
          class=""
        >
          Total
        </p>
        <p
          class="shopping-cart__push-right"
        >
          <span>
            (
            3
            )
          </span>
          <span
            class="underline"
          >
             $
            50
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
